config/pmd.xml

```diff
@@ -104,66 +129,82 @@
-           validateCli is not reasonable to split as encapsulation of logic will be damaged
-           getDetails - huge Switch, it has to be monolithic
-           JavadocMethodCheck, JavadocStyleCheck, JavadocUtils.getJavadocTags() - deprecated
-           FinalLocalVariableCheck.visitToken() - it is just big switch , not could be done
-           GenericWhitespaceCheck.processSingleGeneric() - it contains a lot of similar to each other and very simple condition
-           VariableDeclarationUsageDistanceCheck, CustomImportOrderCheck - it is not OK to have such a complicated logic need to be refactored.
-      -->
-      <property name="violationSuppressXPath" value="//DeclarationOrder[@Name='visitToken'] and //MethodDeclaration[@Name='validateCli' and ../../..[@Image='Main']] | //MethodDeclaration[@Name='processSingleGeneric' and ../../..[@Image='GenericWhitespaceCheck']] | //MethodDeclaration[@Name='visitToken' and ../../..[@Image='FinalLocalVariableCheck']] | //MethodDeclaration[@Name='getJavadocTags' and ../../..[@Image='JavadocUtils']] | //MethodDeclaration[@Name='getDetails' and ../../..[@Image='RightCurlyCheck']] | //ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck' or @Image='JavadocStyleCheck' or @Image='VariableDeclarationUsageDistanceCheck' or @Image='CustomImportOrderCheck']"/>
-    </properties>
-  </rule>
-  <rule ref="rulesets/java/codesize.xml/NPathComplexity">
-    <properties>
-      <!-- JavadocMethodCheck - deprecated
-           WhitespaceAroundCheck.isNotRelevantSituation - npath = 216, but additional extraction
-                                of expression to separate method does not make sense as it will
-                                damage encapsulation of that check.
-      -->
-      <property name="violationSuppressXPath" value="//MethodDeclaration[@Name='isNotRelevantSituation' and ../../..[@Image='WhitespaceAroundCheck']] | //MethodDeclaration[@Name='validateCli' and ../../..[@Image='Main']] | //ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck' or @Image='JavadocStyleCheck']"/>
+      <!-- RequireThisCheck, VariableDeclarationUsageDistanceCheck, XdocsPagesTest,
+             JavadocMethodCheck and CommentsIndentationCheck are big classes and should not
+             be split without a good reason.
+           Main.validateCli is not reasonable to split as encapsulation of the logic will be
+             damaged.
+           *.visitToken are a big SWITCH block with a number of IF blocks.
+             If we split the block to several methods it will damage the readability.
+           JavadocStyleCheck is deprecated.
+           SuppressWarningsHolder.getAcceptableParent, FallThroughCheck.isTerminated,
+             ElementNode.iterateAxis, RootNode.iterateAxis are also huge switches,
+             they has to be monolithic.
+           CustomImportOrderCheck.finishImportList, CustomImportOrderCheck.getImportGroup ???
+           Checker.processFiles ???
+           CheckUtils.parseDouble ??? -->
```

Comment:
https://github.com/checkstyle/checkstyle/blob/90f20e09869c71eb22190ad9c1d46d5deec324a5/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java#L495
https://github.com/checkstyle/checkstyle/blob/90f20e09869c71eb22190ad9c1d46d5deec324a5/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java#L615
https://github.com/checkstyle/checkstyle/blob/90f20e09869c71eb22190ad9c1d46d5deec324a5/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L276
https://github.com/checkstyle/checkstyle/blob/90f20e09869c71eb22190ad9c1d46d5deec324a5/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtils.java#L161

Reply:
Checker.processFiles - there are only 6 decision points in method, not sure why PMD think about it in different way, might be because of "try" or it is nuance/bug in PMD.

CheckUtils.parseDouble - create separate issue on this, it might be good to review necessity of this method at all (bunch of utils methods around), it might be. CC level is 11. Not sure what PMD value it counts, might be bug in PMD. Lets print value that PMD counted and suppress as we have our own check for this.

CustomImportOrderCheck , CC is 10 for them, Not sure what PMD value it counts, Might be bug in PMD.  Lets print value that PMD counted and suppress as we have our own check for this.

Reply:
> CheckUtils.parseDouble

This one is broken:
```
        assertEquals("Invalid parse result", -2.0,
                CheckUtils.parseDouble("-2", TokenTypes.NUM_LONG), 0);
```

fails with
```
java.lang.AssertionError: Invalid parse result 
Expected :-2.0
Actual   :-8.0
```

Reply:
Please create issue to get rid of it.